scio-neo4j/src/main/scala/com/spotify/scio/neo4j/ops/Neo4jSCollectionReadOps.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.neo4j.ops
+
+import com.spotify.scio.coders.{Coder, CoderMaterializer}
+import com.spotify.scio.neo4j.Neo4jOptions
+import com.spotify.scio.neo4j.ops.Neo4jCommon.dataSourceConfiguration
+import com.spotify.scio.values.SCollection
+import magnolify.neo4j.ValueType
+import org.apache.beam.sdk.io.{neo4j => beam}
+import org.apache.beam.sdk.transforms.SerializableFunction
+
+import java.util.{Map => JMap}
+import scala.jdk.CollectionConverters._
+
+/**
+ * Operations for transforming an existing [[SCollection]] via a Neo4j query (read)
+ * @param self
+ *   existing [[SCollection]] (prior to Neo4j read)
+ * @tparam X
+ *   type parameter for existing [[SCollection]]
+ */
+class Neo4jSCollectionReadOps[X](self: SCollection[X]) {
```

Comment:
This is not different from the `Neo4jSCollectionWriteOps`. we should define all extension methods in the them in the same class